nemo/nemo/backends/pytorch/actions.py

```diff
@@ -997,6 +1000,132 @@ def _get_all_modules(
                     for module in callchain:
                         self.modules.add(module[0])
 
+    @staticmethod
+    def __module_export(module,
+                        output,
+                        d_format: DeploymentFormat,
+                        input_example=None,
+                        output_example=None):
+        # Check if output already exists
+        destination = Path(output)
+        if destination.exists():
+            raise FileExistsError(f"Destination {output} already exists. "
+                                  f"Aborting export.")
+
+        input_names = list(module.input_ports.keys())
+        output_names = list(module.output_ports.keys())
+        dynamic_axes = {}
+
+        def __extract_dynamic_axes(port_name: str, ntype: NeuralType,
+                                   dynamic_axes: Dict):
+            if ntype.axis2type is not None and ntype.axis2type != {}:
+                for axis_id, axistype in ntype.axis2type.items():
+                    if issubclass(axistype.semantics, BatchTag) or issubclass(
+                            axistype.semantics, TimeTag):
+                        if port_name not in dynamic_axes:
+                            dynamic_axes[port_name] = []
```

Comment:
Any thoughts about using defaultdict for `dynamic_axes` so we can make this cleaner?

Reply:
can you give example?

Reply:
Such as:
```python
from collections import defaultdict
...
dynamic_axes = defaultdict(list)
# And then we can directly do:
dynamic_axes[port_name].append(axis_id)
```
The defaultdict just gets rid of the need for the check here.

Reply:
changed